examples/move/scratch_off/sources/math.move

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module scratch_off::math {
+
+    /// Hard coding this as 1 billion
+    const MODULO_FACTOR: u256 = 1_000_000_000;
+
+    public fun bytes_to_u256(hashed_beacon: &vector<u8>): u256 {
+        let output = 0;
+        let bytes_length = 32;
+        let idx = 0;
+        while (idx < bytes_length) {
+            let current_byte = *std::vector::borrow(hashed_beacon, idx);
+            output = (output << 8) | (current_byte as u256) ;
+            idx = idx + 1;
+        };
+        output
+    }
+
+    public fun get_result(random_number: u256): u64 {
+        ((random_number % MODULO_FACTOR) as u64)
+    }
+
+    public fun get_random_u64(hashed_beacon: &vector<u8>): u64 {
+        let output: u256 = 0;
+        let bytes_length: u64 = 32;
+        let idx: u64 = 0;
+        while (idx < bytes_length) {
+            let current_byte = *std::vector::borrow(hashed_beacon, idx);
+            output = (output << 8) | (current_byte as u256) ;
+            idx = idx + 1;
+        };
+        ((output % MODULO_FACTOR) as u64)
+    }
+
+    public fun get_random_u64_in_range(hashed_beacon: &vector<u8>, max_range: u64): u64 {
+        let output: u256 = 0;
+        let bytes_length: u64 = 32;
+        let idx: u64 = 0;
+        while (idx < bytes_length) {
+            let current_byte = *std::vector::borrow(hashed_beacon, idx);
+            output = (output << 8) | (current_byte as u256) ;
+            idx = idx + 1;
+        };
+        ((output % MODULO_FACTOR) as u64) % max_range
+    }
+
+    /// Example inputs here are 
+    /// random_number_result = 12340
+    /// prizes_odds = 2500 (25%)
+    /// total_lot = 10000
+    /// 12340 % 10000 = 1234 < 2500 which is true.
+    public fun should_draw_prize(
+        hashed_beacon: &vector<u8>,
+        winning_ticket_amount: u64, 
+        total_tickets: u64
+    ): bool {
+        let random_number_result = get_random_u64(hashed_beacon);
+        // This assumes that mod can hit 0.
```

Comment:
this is ok assuming total_tickets is not very big, say, < 2^32 (which means we still get something that is not perfectly random in [0.. total_tickets], but the bias is small, <2^{-32})

Reply:
But that's assuming that get_random_u64 returns a uniformly random u64, right?

Reply:
right, that's a assuming get_random_u64 returns a random u64

Reply:
total_tickets will be under 3 million or so

Reply:
to be on the safe side, just replace the call to `get_random_u64` with a call to `bytes_to_u256` and use
((random_number_result % (total_tickets as u256)) as u64) < winning_ticket_amount